lib/readyset.rb

```diff
@@ -37,28 +37,19 @@ def self.create_cache!(id: nil, sql: nil, name: nil, always: false)
       raise ArgumentError, 'Exactly one of the `id` and `sql` parameters must be provided'
     end
 
-    query = 'CREATE CACHE '
-    params = []
-
-    if always
-      query += 'ALWAYS '
-    end
-
-    unless name.nil?
-      query += '? '
-      params.push(name)
-    end
-
-    if sql
-      query += 'FROM %s'
-      params.push(sql)
+    suffix = sql ? '%s' : '?'
+    from = (id || sql)
+
+    if always && name
+      Readyset.raw_query('CREATE CACHE ALWAYS ? FROM ' + suffix, name, from)
+    elsif always
+      Readyset.raw_query('CREATE CACHE ALWAYS FROM ' + suffix, from)
+    elsif name
+      Readyset.raw_query('CREATE CACHE ? FROM ' + suffix, name, from)
```

Comment:
The change from `if sql` to `suffix` is fantastic.

Here's an article I found that reflects why `suffix` is an improvement: ["Intention revealing names"](http://wiki.c2.com/?IntentionRevealingNames).